app/ts/utils/augurContractUtils.ts

```diff
@@ -493,3 +493,14 @@ export const getParentUniverse = async (reader: AccountAddress, universe: Accoun
 		args: []
 	})
 }
+
+export const getChildUniverse = async (reader: AccountAddress, universe: AccountAddress, payoutNumerators: EthereumQuantity[], numTicks: bigint, numOutcomes: bigint) => {
+	const PayoutDistributionHash = derivePayoutDistributionHash(payoutNumerators, numTicks, numOutcomes)
+	const client = createReadClient(reader)
+	return await client.readContract({
+		abi: UNIVERSE_ABI,
+		functionName: 'getChildUniverse',
+		address: universe,
+		args: [PayoutDistributionHash]
+	})
```

Comment:
I still feel like these should be memoized.  It feels wrong to parse the ABI every time you call a function.  Also, I thought you could call functions with viem in a way that made it feel more like calling async JS methods?  Is it only micro-eth-signer that can do that?

Reply:
https://viem.sh/docs/contract/getContract